wiki/programming/cmake.md

```diff
@@ -2,11 +2,80 @@
 date: 2017-08-21
 title: CMake and Other Build Systems
 ---
-Build systems are scripts that automate the compilation process. It will be most frequently encountered when compiling software in Linux
-Understanding how they are configured can help you write custom scripts and fix compatibility issues.
+Programs written in compiled languages like C/C++ must be compiled before being run, a process which converts the high-level code written by the programmer into binary files that can be executed on a particular architecture. A build system is a framework that automates the process of compiling files in a project with a specified compiler, and also links predefined libraries of code that are referenced in the project files. By using a build system, a developer can reduce the process of compiling a project to a single line in a terminal, avoiding the redundant steps of managing each file independently.
+
+Popular build tools include Autotools, Make, Maven, Gradle, and Ninja. Each of these frameworks contains a set of protocols for managing project compilation, and the developer can specify the exact requirements, targets, and dependencies for their project. The Make build system refers to these specifications as a Makefile, although other build systems use other terms.
+
+Writing a Makefile can be a complicated task, especially for larger projects. CMake is a tool that simplifies the generation of Makefiles for Make, Android Studio, Ninja, and others. CMake is open-source, and is available for Windows, Mac, and Linux. When a project that uses CMake is built, CMake will first generate a Makefile, and then that file will be used to instruct the build system in executing the project compilation.
+
+When using CMake, it is common and accepted practice to organize a project folder according to a specific folder structure, as seen in the image below.
+
+![cmake_file_structure](/images/2020/12/cmake_file_structure.png)
```

Comment:
Is this file meant to be included? If so, it should be in the `wiki/programming/assets` directory.

Reply:
Thanks, I've fixed that issue.